libkbfs/keybase_daemon_local.go

```diff
@@ -29,40 +29,43 @@ func (m localUserMap) getLocalUser(uid keybase1.UID) (LocalUser, error) {
 }
 
 type favoriteStore interface {
-	FavoriteAdd(folder keybase1.Folder) error
-	FavoriteDelete(folder keybase1.Folder) error
-	FavoriteList(sessionID int) ([]keybase1.Folder, error)
+	FavoriteAdd(uid keybase1.UID, folder keybase1.Folder) error
+	FavoriteDelete(uid keybase1.UID, folder keybase1.Folder) error
+	FavoriteList(uid keybase1.UID) ([]keybase1.Folder, error)
 
 	Shutdown()
 }
 
 type diskFavoriteClient struct {
-	currentUID keybase1.UID
 	favoriteDb *leveldb.DB
 	codec      Codec
 }
 
 var _ favoriteStore = diskFavoriteClient{}
 
-func (c diskFavoriteClient) favkey(folder keybase1.Folder) []byte {
-	return []byte(fmt.Sprintf("%s:%s", c.currentUID, folder.Name))
+func (c diskFavoriteClient) favkey(
+	uid keybase1.UID, folder keybase1.Folder) []byte {
+	return []byte(fmt.Sprintf("%s:%s", uid, folder.ToString()))
```

Comment:
Bug: this was using `folder.Name` instead of `folder.ToString()`, which would conflate public/private folders with the same name.